### ISSUE TITLE
ERROR net.openhft.chronicle.map.ChronicleMap - Checksum doesn't match

### ISSUE THREAD

**Issue by asadganiev** [OpenHFT/Chronicle-Map#198]
Hi,

I am creating ChronicleMap with 250 million entries.

```
ChronicleMap<IntValue, LongValue> pricesByTime = ChronicleMap
                .of(IntValue.class, LongValue.class)
                .name("PRICES_BY_TIME")
                .entries(250_000_000)
                .createOrRecoverPersistedTo(
                        new File(Paths.get(cachePath.toString() +
                                "/" + "PRICES_BY_TIME").toUri()));
```

And closing it before exiting program. But unfortunately each time when I
relaunch or resume program it is printing an error:

```logs
ERROR net.openhft.chronicle.map.ChronicleMap - Checksum doesn't match, stored: 35974103, should be from the entry bytes: -316473655
...
ERROR net.openhft.chronicle.map.ChronicleMap - wrong lowest free chunk for tier with index 3, stored: 31, should be: 35
```

What am I doing wrong and how to fix it?

**Comment by RobAustin** [comment]
could you provide a simple test case that reproduces this issue? It is best if you can submit the test case via a pull request. Thanks

**Comment by asadganiev** [comment]
I am afraid it is a bit hard to reproduce it without showing production code. Because all my tests
pass without any problem. Could it be because of the ChronicleMap was not able to close itself in time before program exits?

**Comment by RobAustin** [comment]
It's going to be very hard for us to fix a problem that can not be reproduced. 

I would suggest on a graceful shutdown that you first close the chronicle map and you should only close the map once all other threads have finished writing to the map.

**Comment by asadganiev** [comment]
This is really good suggestion. It seems some threads keeps putting data after it is ChronicleMap closed. Let me check it. If problem wouldn't be resolved, I will create MCVE, and will pull request.

**Comment by RobAustin** [comment]
Ok - I'll close the issue, for now, let's re-open it later if this does not fix it.

**Comment by hft-team-city** [comment]
Released in [Chronicle-Map-3.17.6](https://github.com/OpenHFT/Chronicle-Map/releases/chronicle-map-3.17.6), [BOM-2.17.396](https://github.com/OpenHFT/OpenHFT/releases/chronicle-bom-2.17.396)

**Comment by hft-team-city** [comment]
Released in [Chronicle-Map-3.17.6](https://github.com/OpenHFT/Chronicle-Map/releases/chronicle-map-3.17.6), [BOM-2.17.407](https://github.com/OpenHFT/OpenHFT/releases/chronicle-bom-2.17.407)
